src/client/dfuse/ops/fgetattr.c

```diff
@@ -1,33 +1,64 @@
 /**
- * (C) Copyright 2016-2022 Intel Corporation.
+ * (C) Copyright 2016-2023 Intel Corporation.
  *
  * SPDX-License-Identifier: BSD-2-Clause-Patent
  */
 
 #include "dfuse_common.h"
 #include "dfuse.h"
 
+static void
+dfuse_cb_getattr_cb(struct dfuse_event *ev)
+{
+	if (ev->de_ev.ev_error != 0) {
+		DFUSE_REPLY_ERR_RAW(ev->de_ie, ev->de_req, ev->de_ev.ev_error);
+		D_GOTO(release, 0);
+	}
+
+	ev->de_attr.st_ino = ev->de_ie->ie_stat.st_ino;
+
+	ev->de_ie->ie_stat = ev->de_attr;
+
+	DFUSE_REPLY_ATTR(ev->de_ie, ev->de_req, &ev->de_attr);
+release:
+	daos_event_fini(&ev->de_ev);
+	D_FREE(ev);
+}
+
 void
 dfuse_cb_getattr(fuse_req_t req, struct dfuse_inode_entry *ie)
 {
-	struct stat	attr = {};
-	int		rc;
+	struct dfuse_info  *dfuse_info = fuse_req_userdata(req);
+	struct dfuse_event *ev;
+	uint64_t            eqt_idx;
+	struct dfuse_eq    *eqt;
+	int                 rc;
 
 	if (ie->ie_unlinked) {
 		DFUSE_TRA_DEBUG(ie, "File is unlinked, returning most recent data");
 		DFUSE_REPLY_ATTR(ie, req, &ie->ie_stat);
 		return;
 	}
 
-	rc = dfs_ostat(ie->ie_dfs->dfs_ns, ie->ie_obj, &attr);
-	if (rc != 0)
-		D_GOTO(err, rc);
+	eqt_idx = atomic_fetch_add_relaxed(&dfuse_info->di_eqt_idx, 1);
+	eqt     = &dfuse_info->di_eqt[eqt_idx % dfuse_info->di_eq_count];
+	D_ALLOC_PTR(ev);
+	if (ev == NULL)
+		D_GOTO(err, rc = ENOMEM);
 
-	attr.st_ino = ie->ie_stat.st_ino;
+	ev->de_req         = req;
+	ev->de_complete_cb = dfuse_cb_getattr_cb;
+	ev->de_ie          = ie;
 
-	ie->ie_stat = attr;
+	rc = daos_event_init(&ev->de_ev, eqt->de_eq, NULL);
+	if (rc != -DER_SUCCESS)
```

Comment:
need to free ev here?

Reply:
Yes, will re-push.

Reply:
you said you will repush to fix this, but sounds like still needs the fix.

Reply:
Ah yes, sorry.  I've got too many PRs in flight at the mintue.